### PR TITLE
SortingPolicyPlugin

### PR DESCRIPTION
**Pull request by andresailer**
BEGINRELEASENOTES
- Plugin: lcgeo_LinearSortingPolicy : new plugin to set the sorting policy for surfaces identified by placement path based on their z position and a linear function, requires AIDASoft/DD4hep#486
- CLIC_o3_v14: add use of lcgeo_LinearSortingPolicy for TrackerEndcaps
- TrackerEndcapSupport_o1_v01: attach support volumes to different DetElements so we can attach different extensions to them

ENDRELEASENOTES

### CODE REVIEW

**Comment by petricm** [comment]
Shouldn't this rather be CLIC_o3_v15?

**Comment by andresailer** [comment]
Except for the split in the support material it does not change any volumes. Anything simulated with previous CLIC_o3_v14 is compatible with the changed xml files.

File: detector/other/TrackerEndcapSupport_o1_v01.cpp [hunk 1]
```diff
@@ -70,6 +70,13 @@ static Ref_t create_detector(Detector& theDetector, xml_h e, SensitiveDetector)
     Tube    l_tub(rmin, rmax, layerWidth, 2 * M_PI);
     Volume  l_vol(l_nam, l_tub, air);
     l_vol.setVisAttributes(theDetector, x_layer.visStr());
+
+    DetElement layer_pos(sdet, l_nam + "_pos", l_num);
+    pv = envelope.placeVolume(l_vol, Position(0, 0, zmin + layerWidth / 2.));
+    pv.addPhysVolID("layer", l_num);
+    pv.addPhysVolID("barrel", 1);
+    layer_pos.setPlacement(pv);
+
```
Comment (eleogran) [inline detector/other/TrackerEndcapSupport_o1_v01.cpp]: what changed here? did you just move this block before the 'for'?

Comment (andresailer) [reply detector/other/TrackerEndcapSupport_o1_v01.cpp]: Yes, so that I could assign the surface to the `layer_pos` DetElement

**Pull request merged**